pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,828 @@
+				.unwrap()
+				.to_string()
+		))
+	);
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn approve_works(mut session: Session) {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy(
+		&mut session,
+		CONTRACT.clone(),
+		"new",
+		vec![TOKEN.to_string(), MIN_BALANCE.to_string()],
+	)
+	.unwrap();
+	session.set_actor(contract.clone());
+	// Mint tokens.
+	assert_ok!(session.sandbox().mint_into(&TOKEN, &contract.clone(), AMOUNT));
+
+	// Successfully approvals.
+	assert_ok!(approve(&mut session, ALICE, AMOUNT / 2));
+	assert_eq!(session.sandbox().allowance(&TOKEN, &contract, &ALICE), AMOUNT / 2);
+	// Successfully emit event.
+	assert_eq!(
+		last_contract_event(&session).unwrap(),
+		Approval {
+			owner: account_id_from_slice(contract.clone().as_ref()),
+			spender: account_id_from_slice(ALICE.as_ref()),
+			value: AMOUNT / 2,
+		}
+		.encode()
+		.as_slice()
+	);
+
+	// No-op if the caller and `spender` is the same address, returns success and no events are
+	// emitted.
+	assert_ok!(approve(&mut session, contract.clone(), AMOUNT / 2));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+	assert_eq!(session.sandbox().allowance(&TOKEN, &contract, &ALICE), AMOUNT / 2);
+
+	// Non-additive, sets new value.
+	assert_ok!(approve(&mut session, ALICE, AMOUNT / 2 - 1));
+	assert_eq!(session.sandbox().allowance(&TOKEN, &contract, &ALICE), AMOUNT / 2 - 1);
+	// Successfully emit event.
+	assert_eq!(
+		last_contract_event(&session).unwrap(),
+		Approval {
+			owner: account_id_from_slice(contract.as_ref()),
+			spender: account_id_from_slice(ALICE.as_ref()),
+			value: AMOUNT / 2 - 1,
+		}
+		.encode()
+		.as_slice()
+	);
+
+	// Token is not live, i.e. frozen or being destroyed.
+	assert_ok!(session.sandbox().start_destroy(&TOKEN));
+	// `pallet-assets` returns `AssetNotLive` error.
+	assert_eq!(
+		approve(&mut session, ALICE, AMOUNT / 2),
+		Err(PSP22Error::Custom(
+			vec_u8_to_u32(Module { index: 52, error: [16, 0] }.encode())
+				.unwrap()
+				.to_string()
+		))
+	);
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn increase_allowance_works(mut session: Session) {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy(
+		&mut session,
+		CONTRACT.clone(),
+		"new",
+		vec![TOKEN.to_string(), MIN_BALANCE.to_string()],
+	)
+	.unwrap();
+	session.set_actor(contract.clone());
+	// Mint tokens and approve.
+	assert_ok!(session.sandbox().mint_into(&TOKEN, &contract.clone(), AMOUNT));
+	assert_ok!(session.sandbox().approve(&TOKEN, &contract.clone(), &ALICE, AMOUNT / 2));
+
+	// Successfully approvals.
+	assert_ok!(increase_allowance(&mut session, ALICE, AMOUNT / 2));
+	assert_eq!(session.sandbox().allowance(&TOKEN, &contract, &ALICE), AMOUNT);
+	// Successfully emit event.
+	assert_eq!(
+		last_contract_event(&session).unwrap(),
+		Approval {
+			owner: account_id_from_slice(contract.as_ref()),
+			spender: account_id_from_slice(ALICE.as_ref()),
+			value: AMOUNT,
```

Comment:
```suggestion
			value,
```

As for the other tests above by adding the variable